src/modules/dropdown.js

```diff
@@ -381,9 +384,12 @@ layui.define(['jquery', 'laytpl', 'lay', 'util'], function(exports){
     
     // 若存在已打开的面板元素，则移除
     if(elemPrev){
-      elemPrev.data('prevElem') && (
-        elemPrev.data('prevElem').data(MOD_INDEX +'_opened', false)
-      );
+      var prevElem = elemPrev.data('prevElem');
+      if(prevElem){
+        prevElem.data(MOD_INDEX +'_opened', false);
+        var prevOpt = prevElem.data(MOD_INDEX +'_on_close');
+        typeof prevOpt.close === 'function' && prevOpt.close(prevElem);
+      }
       elemPrev.remove();
```

Comment:
这里可能是被 `prevElem` 误导了，直接操作原型链是否也能达到目的呢，晨星可做进一步验证。

```suggestion
      var prevElem = elemPrev.data('prevElem');
      if (prevElem) {
        prevElem.data(MOD_INDEX +'_opened', false);
      }
      elemPrev.remove();
      delete thisModule.prevElem;
      typeof options.close === 'function' && options.close(prevElem);
```

Reply:
如果直接操作原型链，以下场景将无法正常触发 close 事件
1. 点击 A, A 打开
2. 鼠标悬停 B，A 关闭，B 打开

此时 A 关闭调用的是 B 的 close 事件
```  js
dropdown.render({
    elem: '#demoA'
    ,content: '自定义内容  123 '
    ,trigger: 'click'
    ,close: function(){
      console.log('demoA', this.elem);
    }
});
dropdown.render({
    elem: '#demoB'
    ,content: '自定义内容  123 '
    ,trigger: 'hover'
    ,close: function(){
      console.log('demoB', this.elem);
    }
});
```